fedimint-api-client/src/api/global_api.rs

```diff
@@ -129,12 +132,56 @@ where
         .map_err(|e| anyhow!(e.to_string()))
     }
 
+    fn select_peers_for_status(&self) -> impl Iterator<Item = PeerId> + '_ + Clone {
+        // TODO: better selection
+        self.all_peers().iter().cloned()
+    }
+
     async fn get_session_status_raw(
         &self,
         block_index: u64,
         decoders: &ModuleDecoderRegistry,
     ) -> anyhow::Result<SessionStatus> {
         debug!(target: LOG_CLIENT_NET_API, block_index, "Fetching block's outcome from Federation");
+        let verify = |_: &SignedSessionOutcome| -> bool { todo!() };
+        // TODO: check api version
+        // fetch serially
+        for peer_id in self.select_peers_for_status().cycle() {
+            // TODO: maybe even multiple retries with same peer.
+            match self
+                .request_single_peer_federation::<SerdeModuleEncoding2<SessionStatusV2>>(
+                    SESSION_STATUS_V2_ENDPOINT.to_string(),
+                    ApiRequestErased::new(block_index),
+                    peer_id,
+                )
+                .await
+                .map_err(anyhow::Error::from)
+                .and_then(|s| Ok(s.try_into_inner(decoders)?))
+            {
+                Ok(SessionStatusV2::Complete(signed_session_outcome)) => {
+                    if verify(&signed_session_outcome) {
+                        // early return
+                        return Ok(SessionStatus::Complete(
+                            signed_session_outcome.session_outcome,
+                        ));
+                    } else {
+                        // evil peer, try with next peer
+                        // TODO: decrease score of this peer to avoid selecting it again for future
+                        // requests.
+                        continue;
+                    }
+                }
+                Ok(SessionStatusV2::Initial | SessionStatusV2::Pending(..)) => {
+                    // ask other peers
+                    // use fallback method
+                    break;
+                }
+                Err(_) => {
+                    // try next peer
+                    continue;
+                }
+            }
+        }
         self.request_current_consensus::<SerdeModuleEncoding<SessionStatus>>(
```

Comment:
Is there any point keeping now?

Reply:
we still need current consensus for pending and initial values because those don't have signature

Reply:
Oh, right.